src/main/kotlin/com/pubnub/api/presence/eventengine/PresenceEventEngine.kt

```diff
@@ -0,0 +1,42 @@
+package com.pubnub.api.presence.eventengine
+
+import com.pubnub.api.eventengine.EventEngine
+import com.pubnub.api.eventengine.Sink
+import com.pubnub.api.eventengine.Source
+import com.pubnub.api.eventengine.transition
+import com.pubnub.api.presence.eventengine.effect.PresenceEffectInvocation
+import com.pubnub.api.presence.eventengine.event.PresenceEvent
+import com.pubnub.api.presence.eventengine.state.PresenceState
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+class PresenceEventEngine(
```

Comment:
Generalized EventEngine implementation should be in `c.p.a.eventengine` package and both Subscribe and Presence should use it. Both have exactly the same implementation so there's no need to duplicate it